migration/v1_to_v2/data/exporters/data_exporter.rb

```diff
@@ -143,14 +143,8 @@ def value_to_ruby_string(value, include_blank = false)
   # rubocop:enable Metrics/MethodLength
   # rubocop:enable Metrics/AbcSize
 
-  def valid_key?(key)
-    return false if key.is_a?(Integer) || key.include?('-') || key.include?(' ') || !key.match?('^[a-zA-Z]')
-
-    true
-  end
-
   def key_to_ruby(key)
-    valid_key?(key) ? key : "'#{key}'"
+    "'#{ key.gsub(/'/, "\\'") }'"
```

Comment:
If you receive an integer you will get an exception. so you can keep the validation and add your escape to those that are not valid:

```
def key_to_ruby(key)
  return key if valid_key?(key)

  "'#{ key to_s.gsub(/'/, "\\'") }'"
end
```

Reply:
I wouldn't want to rely on any `valid_key?` method for this, everything we're being passed should be a valid key given that it comes from an existing ruby `key: value` pair afaik.

I would instead just coerce everything to string before applying the `.gsub`.